logbook-spring-boot-autoconfigure/src/main/java/org/zalando/logbook/autoconfigure/LogbookAutoConfiguration.java

```diff
@@ -377,4 +383,31 @@ private static FilterRegistrationBean newFilter(final Filter filter, final Strin
         return registration;
     }
 
+    @Configuration(proxyBeanMethods = false)
+    @ConditionalOnClass(reactor.netty.http.server.HttpServer.class)
+    @ConditionalOnWebApplication(type = REACTIVE)
+    static class WebFluxServerConfiguration {
+
+        static final String CUSTOMIZER_NAME = "logbookNettyServerCustomizer";
+
+        @Bean
+        @ConditionalOnProperty(name = "logbook.filter.enabled", havingValue = "true", matchIfMissing = true)
+        @ConditionalOnMissingBean(name = CUSTOMIZER_NAME)
+        public NettyServerCustomizer logbookNettyServerCustomizer(final Logbook logbook) {
+            return httpServer -> httpServer.tcpConfiguration(tcpServer -> tcpServer.doOnConnection(connection -> connection.addHandlerLast(new LogbookServerHandler(logbook))));
+        }
```

Comment:
Seems like this does not work nicely together with spring-cloud-starter-sleuth (2.2.5), specifically traceId & spanId MDC variables are empty in logbook's request & response lines.
Being new to webflux, I don't have any ideas how to fix this, but thought it's probably worth sharing 🤔

Reply:
> MDC variables

MDC is just a fancy name for a thread-local map. ThreadLocal in general doesn't work with reactive systems (Netty, Reactor, Rx, etc) because their approach to multi-threading does not guarantee a that the same thread is being used.

- https://projectreactor.io/docs/core/release/reference/#faq.mdc
- https://www.novatec-gmbh.de/en/blog/how-can-the-mdc-context-be-used-in-the-reactive-spring-applications/
- https://github.com/reactor/reactor-core/issues/1985

It's an issue that is not directly related to Logbook, but more to the very nature of reactive systems. 
I have to admit, that I'm not fully in the topic, so if someone has productive suggestions and contributions to Logbook that help with that in any way, I'm very much interested in that.

Reply:
@sergey-volkov-lm Can you take this discussion to a separate issue, if you're still interested in that? It's not too strictly related to this PR.

Reply:
@whiskeysierra about sleuth & MDC: it claims support for reactive applications, seems to work.
I suppose that in order to integrate with sleuth (and spring reactive ecosystem in general) one have to implement WebFilter, so this relates to spring-auto-configuration. Would you like me to move this thread into original issue #740 or create new one?

Reply:
> Do you have some links/references?

(docs) [sleuth integrations: reactor](https://docs.spring.io/spring-cloud-sleuth/docs/current-SNAPSHOT/reference/html/integrations.html#sleuth-reactor-integration)
(source code) [Sleuth's TraceWebFilter](https://github.com/spring-cloud/spring-cloud-sleuth/blob/master/spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceWebFilter.java)
(stackoverflow) [question about logging request/response bodies answered by Spring member](https://stackoverflow.com/a/45280764/6629674)

Reply:
If I'm reading those correctly, then you can actually fix those outside of Logbook by augmenting/configuring sleuth or their filter differently. I'd prefer to create a completely new issue for this.